File: 알고리즘/2/Determine if String Halves Are Alike .py
class Solution:
    basic ='aeiouAEIOU'
    
    def cnt(self, s):

        return len([[ele for ele in s if ele in Solution.basic]])

    def halvesAreAlike(self, s: str) -> bool:
        
        a, b = self.cnt(str(s[:len(s)//2])), self.cnt(str(s[len(s)//2:]))
        
        return a==b

File: algorithms/2/Determine if String Halves Are Alike.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
  string basic = "aeiouAEIOU";
    int cnt(string s){
      int count = 0;
      for(auto ele : s){
        if(basic.find(ele)!=string::npos) count+=1;
      }
      return count;
    }

    bool halvesAreAlike(string s) {
      
      string a, b;

      for (int i = 0; i < s.size(); i++)
      {
        if(i<s.size()/2){
          a+=s[i];
        } else{
          b+=s[i];
        }
      }
      int acount = cnt(a);
      int bcount = cnt(b);

      return acount==bcount;
    }
};